### PR TITLE
[diffusion, tests] feat: add BOOGU training adapters for CPU-tested paths

### PR DESCRIPTION
**Pull request by zyforsure**
### What does this PR do?

Add initial BOOGU diffusion training support on the CPU-tested path.

This PR introduces BOOGU training adapters for `flow_grpo` and `diffusion_nft`, plus the minimum shared compatibility changes required to exercise them in CPU-side tests.

Specifically, this PR:
- adds `BooguImagePipeline` / `BooguImageTurboPipeline` training adapters for `flow_grpo`
- adds `BooguImagePipeline` / `BooguImageTurboPipeline` training adapters for `diffusion_nft`
- adds shared BOOGU helper logic for time-shifted sigma construction, rotary embeddings, CFG application, and condition-image normalization
- allows unified I2I adapters to degenerate to a T2I path when condition images are absent
- extends request parsing to accept BOOGU-style `instruction` / `negative_instruction` payloads
- makes top-level optional imports tolerant of missing rollout-only dependencies so CPU-side adapter tests can run without a full `vllm` / `vllm-omni` environment

This PR is intentionally scoped to the training-side / CPU-tested path. It does **not** claim end-to-end BOOGU rollout support is complete yet.

### Checklist Before Starting

- [x] Search for similar PRs. Paste at least one query link here:
  - https://github.com/verl-project/verl-omni/pulls?q=boogu
  - https://github.com/verl-project/verl-omni/pulls?q=diffusion+adapter
  - https://github.com/verl-project/verl-omni/pulls?q=instruction+negative_instruction
- [x] Format the PR title as `[{modules}] {type}: {description}` (This will be checked by the CI)

### Test

Targeted CPU-side tests:

```bash
python -m pytest
tests/pipelines/test_i2i_model_base_on_cpu.py
tests/pipelines/test_image_edit_interface_on_cpu.py
tests/pipelines/test_boogu_image_adapters_on_cpu.py
```
Result:

```text
33 passed
```
Additional validation:
- `python -m py_compile` on the added / updated BOOGU adapter files
- local git diff sanity check before commit

Notes:
- This PR only validates the BOOGU training-side adapter path on CPU.
- End-to-end BOOGU rollout with `vllm` / `vllm-omni` is intentionally left out of this PR.
-This PR is one step toward the full feature. To make review easier, it is intentionally limited in scope.
### API and Usage Example

No public API change.

Existing request parsing and adapter resolution entrypoints remain the same. This PR only extends accepted payload keys and registered diffusion architectures.
### Reference Training Results (Smoke)

A minimal smoke run was executed  to validate the BOOGU training-side adapter wiring。

- steps: 60
- loss: 2.26e-2 → 6e-6
loss_curve

### CODE REVIEW

**Comment by SamitHuang** [comment]
@zhtmike PTAL

**Review by SamitHuang — COMMENTED** [review]
Nice work. Pls attach the reference training results to verify the implementation is correct

**Comment by zyforsure** [comment]
> Nice work. Pls attach the reference training results to verify the implementation is correct

I have finished one smoke test on this，please check it.

File: scripts/run_boogu_adapter_smoke.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+
+def build_model_config(true_cfg_scale: float = 2.0):
+    cfg = SimpleNamespace()
+    cfg.algorithm = "flow_grpo"
+    cfg.architecture = "BooguImagePipeline"
+    cfg.external_lib = None
+    cfg.pipeline = SimpleNamespace(height=64, width=64, num_inference_steps=4, true_cfg_scale=true_cfg_scale)
+    cfg.algo = SimpleNamespace(noise_level=0.0, sde_type="sde")
+    cfg.local_path = "DUMMY/BOOGU"
```
Comment (SamitHuang) [inline scripts/run_boogu_adapter_smoke.py:20]: We should verify with the training pipeline with the complete BOOGU model weights

File: scripts/run_boogu_adapter_smoke.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+            prompt_embeds_mask=prompt_mask,
+            negative_prompt_embeds=negative_prompt_embeds,
+            negative_prompt_embeds_mask=negative_prompt_mask,
+            micro_batch=micro_batch,
+            step=0,
+        )
+        prediction = BooguImageFlowGRPO.forward(module, model_config, model_inputs, negative_model_inputs)
+        loss = (prediction**2).mean()
+        loss.backward()
+        optimizer.step()
```
Comment (SamitHuang) [inline scripts/run_boogu_adapter_smoke.py:92]: This MSE overfiting test shows the tiny BOOGU forward and backprop are ok. But we care more on ensuring the model with the FlowGRPO/NFT (e2e training) is runnable and convergent. A better smoking test should be like tests/gpu_smoke/run_gpu_smoke_diffusion_e2e.sh

**Review by SamitHuang — COMMENTED** [review]
To be concrete, we need an e2e training test to verifying the correctness and covergence when integrating a new model. I'd suggest to run further training tests referring to QwenImage flowgrpo/nft experiment (examples/flowgrpo_trainer/README.md)

**Comment by zhtmike** [comment]
> To be concrete, we need an e2e training test to verifying the correctness and covergence when integrating a new model. I'd suggest to run further training tests referring to QwenImage flowgrpo/nft experiment (examples/flowgrpo_trainer/README.md)

Agree.

**Comment by SamitHuang** [comment]
close as #332 merged. thanks for the exploration!
